spec/concepts/jwt_api_entreprise/operation/notify_expiration_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+describe JwtApiEntreprise::Operation::NotifyExpiration do
+  describe 'expire_in: option (provide a number of days)' do
+    let(:user_1) { create(:user) }
+    let(:user_2) { create(:user) }
+    let(:user_3) { create(:user) }
+
+    # Let's test with a 90 days (3 months) expiration notice
+    # Two tokens on the three are targetted
+    let(:days) { 90 }
+
+    let!(:jwt_1) { create(:jwt_expiring_in_3_month, user: user_1) }
+    let!(:jwt_2) { create(:jwt_expiring_in_3_month, user: user_2) }
+    let!(:jwt_3) { create(:jwt_expiring_in_1_year, user: user_3) }
+
+    subject { described_class.call(expire_in: days) }
+
+    it { is_expected.to be_success }
+
+    it 'calls the mailer for the affected JWT only' do
+      expect(JwtApiEntrepriseMailer).to receive(:expiration_notice).with(jwt_1, days).and_call_original
+      expect(JwtApiEntrepriseMailer).to receive(:expiration_notice).with(jwt_2, days).and_call_original
+      expect(JwtApiEntrepriseMailer).to_not receive(:expiration_notice).with(jwt_3, days).and_call_original
+
+      subject
+    end
+
+    it 'does not send the same notification twice' do
+      described_class.call(expire_in: days)
+      expect(JwtApiEntrepriseMailer).to_not receive(:expiration_notice).with(jwt_1, days)
+      expect(JwtApiEntrepriseMailer).to_not receive(:expiration_notice).with(jwt_2, days)
+
+      subject
+    end
+
+    it 'saves the notification has been sent' do
```

Comment:
remembers sent notifications (shorter and more idiomatic english)

Reply:
records can replace remember as a verb

Reply:
En général j'utilise "save" pour indiquer la persistance en base car ça fait echo à la méthode d'ActiveRecord, et j'aimerai bien garder "save" car je l'utilise partout. Je te laisse faire un ag pour voir 😅 

Le terme "records" il m'arrive de l'utiliser pour désigner une instance de modèle.

Reply:
Je corrige la typo.

Reply:
Ok pour la consistance 👍